diode/webclient/renderer.js

```diff
@@ -205,6 +205,161 @@ class CanvasManager {
+            // If the parent_id isn't null and there is a parent graph, we can
+            // look up the parent node via the element's parent_id
+            parent_element = parent_graph.node(el.parent_id);
+            // If our parent element is a state, we want the state's graph
+            if (parent_element.data.state)
+                parent_graph = parent_element.data.graph;
+        }
+
+        if (parent_graph) {
+            // Find all the edges connected to the moving node
+            parent_graph.outEdges(el.id).forEach(edge_id => {
+                out_edges.push(parent_graph.edge(edge_id));
+            });
+            parent_graph.inEdges(el.id).forEach(edge_id => {
+                in_edges.push(parent_graph.edge(edge_id));
+            });
+        }
+
+        if (parent_element) {
+            // Calculate the box to bind the element to. This is given by
+            // the parent element, i.e. the element where out to-be-moved
+            // element is contained within
+            const min_x = (parent_element.x - (parent_element.width / 2)) +
+                (el.width / 2);
+            const min_y = (parent_element.y - (parent_element.height / 2)) +
+                (el.height / 2);
+            const max_x = (min_x + parent_element.width) - el.width;
+            const max_y = (min_y + parent_element.height) - el.height;
+
+            // Make sure we don't move our element outside its parent's
+            // bounding box
+            const target_x = el.x + dx;
+            const target_y = el.y + dy;
+            if (target_x <= min_x || target_x >= max_x) dx = 0;
+            if (target_y <= min_y || target_y >= max_y) dy = 0;
+        }
+
+        if (el.data.type && (el.data.type === 'Memlet' || el.data.type === 'InterstateEdge')) {
+            if (el.points[2]) {
+                // Only allow dragging, if the memlet is 'making a curve'
+                el.points[1].x += dx;
+                el.points[1].y += dy;
+            }
+            // Don't do any of the other stuff, it doesn't apply here
+            return;
+        }
+
+        // Move a node together with its connectors if it has any
+        function move_node_and_connectors(node) {
+            node.x += dx;
+            node.y += dy;
+            if (node.data.node && node.data.node.type === 'NestedSDFG')
+                translate_recursive(node.data.graph);
+            if (node.in_connectors)
+                node.in_connectors.forEach(c => {
+                    c.x += dx;
+                    c.y += dy;
+                });
+            if (node.out_connectors)
+                node.out_connectors.forEach(c => {
+                    c.x += dx;
+                    c.y += dy;
+                });
+        }
+
+        // Allow recursive translation of nested SDFGs
+        function translate_recursive(ng) {
+            ng.nodes().forEach(state_id => {
+                const state = ng.node(state_id);
```

Comment:
why not use `let`?

Reply:
I assumed `prefer-const`, is let generally preferred?

Reply:
you can use const if it is correct. I normally don't.